### ISSUE TITLE
"Align logical expression" - Needless Spaces

### ISSUE THREAD

**Issue by hypo1989** [SAP/abap-cleaner#232]
The Equal-Sign is moved, although the IS NOT INITIAL is not aligned, because the limit of inner spaces is exceeded.

![image](https://github.com/SAP/abap-cleaner/assets/157125669/d6834990-cab9-46ad-be2a-db3bdec626fd)

The limit ist not exceeded it look right.
![image](https://github.com/SAP/abap-cleaner/assets/157125669/2ee60115-d37e-4620-9e7a-b27120d1b2cd)

**Comment by jmgrassau** [comment]
Hi hypo1989, 

this option was actually meant that way! Think of a more extreme example: 

![image](https://github.com/SAP/abap-cleaner/assets/79639298/56a211f0-a36c-463e-96fc-96534b05d0c3)

I guess you wouldn't want the first comparison operator to be aligned, because you could hardly realize that the line goes on after ```is_buffer-id``` …

```ABAP
  CHECK     is_buffer-id                           > 3
        AND is_buffer-any_flag_with_long_name      = abap_false
        AND is_buffer-other_flag_with_long_name    = abap_false
        AND is_buffer-third_flag_with_longest_name = abap_true.
```

… but it would be a pity if that meant that none of them could be aligned:

```ABAP
  CHECK     is_buffer-id > 3
        AND is_buffer-any_flag_with_long_name = abap_false
        AND is_buffer-other_flag_with_long_name = abap_false
        AND is_buffer-third_flag_with_longest_name = abap_true.
```

Therefore, the option changes everything that can be aligned without inserting too many spaces: 

```ABAP
  CHECK     is_buffer-id > 3
        AND is_buffer-any_flag_with_long_name      = abap_false
        AND is_buffer-other_flag_with_long_name    = abap_false
        AND is_buffer-third_flag_with_longest_name = abap_true.
```

Kind regards,
Jörg-Michael

**Comment by hypo1989** [comment]
Yes but i think in line 4 and 5 there is one blank too many inserted. It depends on the IS NOT INITIAL.
![image](https://github.com/SAP/abap-cleaner/assets/157125669/4060918d-631a-4f0b-ab22-45afd5b7e690)
